src/formats/onenote.ts

```diff
@@ -0,0 +1,467 @@
+				// If it's any other website, convert to a basic link
+				const linkNode = document.createElement('a');
+				linkNode.href = video.src;
+				video.parentNode?.replaceChild(linkNode, video);
+			}
+		});
+		return pageElement;
+	}
+
+	// Downloads attachments from the attachmentQueue once the file has been created.
+	async fetchAttachmentQueue(progress: ImportContext, currentFile: TFile) {
+		if (this.attachmentQueue.length >= 1) {
+			let attachmentPath: string = (await this.getOutputFolder())!.path + '/OneNote Attachments';
+
+			// @ts-ignore
+			// Bug: This function always returns the path + "Note name.md" rather than just the path for some reason
+			if (this.useDefaultAttachmentFolder) attachmentPath = await this.app.vault.getAvailablePathForAttachments(currentFile.basename, currentFile.extension, currentFile);
+	
+			// Create the attachment folder if it doesn't exist yet
+			try {
+				this.vault.createFolder(attachmentPath);
+			}
+			catch (e) { }
+	
+			this.attachmentQueue.forEach(async attachment => {
+				try {
+					const data = (await this.graphHelper.requestUrl(attachment.contentLocation!, 'file')) as ArrayBuffer;
+					await this.app.vault.createBinary(attachmentPath + '/' + attachment.name, data);
+		
+					progress.reportAttachmentSuccess(attachment.name!);	
+				}
+				catch (e) {
+					progress.reportFailed(attachment.name!, e);
+				}
+			});
+	
+			// Clear the attachment queue after every note
+			this.attachmentQueue = [];		
+		}
+		else { }
+	}
+	
+	// Convert OneNote styled elements to valid HTML for proper htmlToMarkdown conversion
+	styledElementToHTML(pageElement: HTMLElement): HTMLElement {
+		const styledElements = pageElement.querySelectorAll('[style]');
+		
+		// For some reason cites/quotes are not converted into Markdown (possible htmlToMarkdown bug), so we do it ourselves temporarily
+		const cites = pageElement.findAll('cite');
+		cites.forEach((cite) => cite.innerHTML = '> ' + cite.innerHTML + '<br>');
+
+		// Map styles to their elements
+		const styleMap: { [key: string]: string } = {
+			'font-weight:bold': 'b',
+			'font-style:italic': 'i',
+			'text-decoration:underline': 'u',
+			'text-decoration:line-through': 's',
+			'background-color': 'mark',
+			'font-family:Consolas': 'pre',
+		};
+
+		styledElements.forEach(element => {
+			const style = element.getAttribute('style') || '';
+			const matchingStyle = Object.keys(styleMap).find(key => style.includes(key));
+	
+			if (matchingStyle) {
+				const newElementTag = styleMap[matchingStyle];
+				const newElement = document.createElement(newElementTag);
+
+				if (newElementTag === 'pre') {
+					const code = newElement.createEl('code');
+					code.textContent = element.textContent;
+				}
+
+				else newElement.innerHTML = element.innerHTML;
+				element.replaceWith(newElement);
+			}
+		});
+
+		return pageElement;
+	}
+	
+	/* Commented out for possible future use. As of now, it seems like it's htmlToMarkdown's fault rather than OneNote's
```

Comment:
Can you explain what's wrong here? Someone reading this code has no idea what this is intended to fix or what's wrong with htmlToMarkdown.